### PR TITLE
Extract inner classes from metrics API

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: api/metrics/src/main/java/io/opentelemetry/api/metrics/DoubleCounterBound.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.metrics;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/** A {@code Bound Instrument} for a {@link DoubleCounter}. */
+@ThreadSafe
+public interface DoubleCounterBound extends BoundSynchronousInstrument {
```
Comment (jkwatson) [inline api/metrics/src/main/java/io/opentelemetry/api/metrics/DoubleCounterBound.java:12]: why the rename? I think `BoundDoubleCounter` was a better name.

**Comment by jkwatson** [comment]
ok, aside from renaming all the bound instrument classes. I much prefer having the "bound" adjective at the front of the class name.

File: api/metrics/src/main/java/io/opentelemetry/api/metrics/DoubleCounterBound.java (hunk 1 above)
Comment (bogdandrutu) [reply api/metrics/src/main/java/io/opentelemetry/api/metrics/DoubleCounterBound.java:12]: Was mostly trying to have classes together grouped by name. I understand your concern, will fix.

Comment (bogdandrutu) [reply api/metrics/src/main/java/io/opentelemetry/api/metrics/DoubleCounterBound.java]: Done.

**Review by jkwatson — APPROVED** [review]
Thanks!

**Pull request merged**